pkg/cache/azure_cache.go

```diff
@@ -157,6 +159,45 @@ func (t *TimedCache) Get(key string, crt AzureCacheReadType) (interface{}, error
 	return entry.Data, nil
 }
 
+// Get returns the requested item by key with deep copy.
+func (t *TimedCache) GetWithDeepCopy(key string, crt AzureCacheReadType) (interface{}, error) {
```

Comment:
Would it be better to decouple `CopyObject` and `Get` since `Get` and `GetWithDeepCopy` share most of their code?

```go
v, _ := cache.Get(...)

// If need mutation
cv := xx.CopyObject(v)
```

Reply:
Good idea. But now I use `Get()` for all references. PTAL